File: app/src/main/java/de/christinecoenen/code/zapp/app/settings/ui/SettingsActivity.java
package de.christinecoenen.code.zapp.app.settings.ui;

import android.content.Context;
import android.content.Intent;
import android.support.v7.app.ActionBar;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.MenuItem;

public class SettingsActivity extends AppCompatActivity {


	public static Intent getStartIntent(Context context) {
		return new Intent(context, SettingsActivity.class);
	}

	@Override
	protected void onCreate(Bundle savedInstanceState) {
		super.onCreate(savedInstanceState);

		ActionBar actionBar = getSupportActionBar();
		if (actionBar != null) {
			actionBar.setDisplayHomeAsUpEnabled(true);
		}

		// display the fragment as the main content.
		getFragmentManager().beginTransaction()
			.replace(android.R.id.content, SettingsFragment.newInstance())
			.commit();
	}

	@Override
	public boolean onOptionsItemSelected(MenuItem item) {
		switch (item.getItemId()) {
			case android.R.id.home:
				finish();
				return true;
		}

		return super.onOptionsItemSelected(item);
	}
}

File: app/src/main/java/de/christinecoenen/code/zapp/app/settings/ui/SettingsActivity.kt
package de.christinecoenen.code.zapp.app.settings.ui

import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.view.MenuItem
import androidx.appcompat.app.AppCompatActivity
import de.christinecoenen.code.zapp.app.settings.ui.SettingsFragment.Companion.newInstance

class SettingsActivity : AppCompatActivity() {

	companion object {

		@JvmStatic
		fun getStartIntent(context: Context): Intent {
			return Intent(context, SettingsActivity::class.java)
		}

	}

	override fun onCreate(savedInstanceState: Bundle?) {
		super.onCreate(savedInstanceState)

		supportActionBar?.setDisplayHomeAsUpEnabled(true)

		// display the fragment as the main content.
		supportFragmentManager.beginTransaction()
			.replace(android.R.id.content, newInstance())
			.commit()
	}

	override fun onOptionsItemSelected(item: MenuItem): Boolean {
		return when (item.itemId) {
			android.R.id.home -> {
				finish()
				true
			}
			else -> {
				super.onOptionsItemSelected(item)
			}
		}
	}
}
